source/logrepl/snapshot.go

```diff
@@ -0,0 +1,219 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logrepl
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/conduitio/conduit-connector-postgres/pgutil"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+)
+
+var psql = sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
+
+const actionSnapshot string = "snapshot"
+
+type SnapshotConfig struct {
+	SnapshotName string
+	Table        string
+	Columns      []string
+	KeyColumn    string
+}
+
+type SnapshotIterator struct {
+	config SnapshotConfig
+
+	tx   pgx.Tx
+	rows pgx.Rows
+
+	internalPos int64
+}
+
+func NewSnapshotIterator(ctx context.Context, conn *pgx.Conn, cfg SnapshotConfig) (*SnapshotIterator, error) {
+	s := &SnapshotIterator{
+		config: cfg,
+	}
+
+	err := s.startSnapshotTx(ctx, conn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start snapshot tx: %w", err)
+	}
+
+	err = s.loadRows(ctx)
+	if err != nil {
+		if rollErr := s.tx.Rollback(ctx); err != nil {
+			sdk.Logger(ctx).Err(err).Msg("load rows failed")
+			return nil, fmt.Errorf("rollback failed: %w", rollErr)
```

Comment:
Nitpick: I'd rather turn the logic around and just log the rollback error (same as we do in line 107)
```suggestion
			sdk.Logger(ctx).Err(rollErr).Msg("rollback failed")
```